python_modules/libraries/dagster-dg/dagster_dg/cli/global_options.py

```diff
@@ -0,0 +1,67 @@
+from pathlib import Path
+from typing import Any, Callable, Optional, Sequence, TypeVar, Union
+
+import click
+
+from dagster_dg.config import DgConfig
+
+T_Command = TypeVar("T_Command", bound=Union[Callable[..., Any], click.Command])
+
+# Defaults are defined on the DgConfig object.
+GLOBAL_OPTIONS = {
+    option.name: option
+    for option in [
+        click.Option(
+            ["--cache-dir"],
+            type=Path,
+            default=DgConfig.cache_dir,
+            help="Specify a directory to use for the cache.",
+        ),
+        click.Option(
+            ["--disable-cache"],
+            is_flag=True,
+            default=DgConfig.disable_cache,
+            help="Disable the cache..",
+        ),
+        click.Option(
+            ["--verbose"],
+            is_flag=True,
+            default=DgConfig.verbose,
+            help="Enable verbose output for debugging.",
+        ),
+        click.Option(
+            ["--builtin-component-lib"],
+            type=str,
+            default=DgConfig.builtin_component_lib,
+            help="Specify a builitin component library to use.",
+        ),
+    ]
+}
+
+
+def dg_global_options(
+    fn: Optional[T_Command] = None, *, names: Optional[Sequence[str]] = None
+) -> Union[T_Command, Callable[[T_Command], T_Command]]:
```

Comment:
you could potentially change this to return a function that handles the call to `DgContext.from_cli_global_options(global_options)` on the users behalf and have the user-provided callback operate on a `DgContext`

Reply:
Would strongly prefer to avoid this since it alters the fundamental API of `click`. `dg_global_options` is designed to be popped in with other click options, would rather keep our callback's click-standard.